mesonbuild/wrap/wrap.py

```diff
@@ -525,8 +543,20 @@ def get_data(self, urlstring: str) -> T.Tuple[str, str]:
         elif WHITELIST_SUBDOMAIN in urlstring:
             raise WrapException(f'{urlstring} may be a WrapDB-impersonating URL')
         else:
+            headers = {'User-Agent': f'mesonbuild/{coredata.version}'}
+            creds = self.get_netrc_credentials(url.netloc)
+
+            req_urlstring = urlstring
+            if creds is not None and '@' not in url.netloc:
+                login, password = creds
+                if url.scheme == 'https':
+                    enc_creds = b64encode(f'{login}:{password}'.encode()).decode()
+                    headers.update({"Authorization": f"Basic {enc_creds}"})
+                else:
+                    req_urlstring = urllib.parse.urlunparse(url._replace(netloc=f"{login}:{password}@{url.netloc}"))
```

Comment:
I think you want to restrict this to ftp? This will now run in the http-without-ssl case.

Reply:
Weirder:
> As is the case with all named tuples, the subclass has a few additional methods and attributes that are particularly useful. One such method is _replace(). The _replace() method will return a new ParseResult object replacing specified fields with new values.